For issue #18132 update to proton icons first batch

Amejia481 opened the pull request:
Updating new icons set, for now updating only :

- [ ] Settings
- [ ] Search
- [ ] Tracking protection enabled
- [ ] Tracking protection disabled
- [ ] Tracking protection blocking trackers
- [ ] Add-ons
- [ ] Info
- [ ] Private Browsing
- [ ] Bookmark
- [ ] Arrowhead right
- [ ] bookmark filled
- [ ] bookmark outline
- [ ] Clear
- [ ] Close
- [ ] Delete
- [ ] desktop
- [ ] Download
- [ ] Synced tabs
- [ ] Tabs counter
- [ ] Whats new

From AC
- [ ] Back
- [ ] Suggestions
 - [ ] Add-on
 - [ ] Forward
 - [ ] Link
 - [ ] Lock
 - [ ] Reader mode
 - [ ] Refresh
 - [ ] Stop
 - [ ] Sync 
 - [ ] Warning 



https://user-images.githubusercontent.com/773158/115661688-d812a380-a30b-11eb-86c1-6ee2becf1a9f.mp4




### Pull Request checklist
<!-- Before submitting the PR, please address each item -->
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

Amejia481 commented:
Fixed tracking protection animations when blocking trackers!

gabrielluong commented on app/src/main/res/drawable/ic_proton_search.xml:
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?><!-- This Source Code Form is subject to the terms of the Mozilla Public
```
```suggestion
<?xml version="1.0" encoding="utf-8"?>
<!-- This Source Code Form is subject to the terms of the Mozilla Public
```

for most of these

gabrielluong commented on app/src/main/res/drawable/ic_proton_settings.xml:
```diff
@@ -0,0 +1,15 @@
+    <path
+        android:fillColor="?primaryText"
+        android:pathData="M12,15.5C10.07,15.5 8.5,13.93 8.5,12C8.5,10.07 10.07,8.5 12,8.5C13.93,8.5 15.5,10.07 15.5,12C15.5,13.93 13.93,15.5 12,15.5ZM12,10C10.897,10 10,10.897 10,12C10,13.103 10.897,14 12,14C13.103,14 14,13.103 14,12C14,10.897 13.103,10 12,10Z" />
+</vector>
```
Add a new line EOF. Check the others

gabrielluong commented on app/src/main/java/org/mozilla/fenix/ext/ProtonIcons.kt:
```diff
@@ -0,0 +1,106 @@
+
+import org.mozilla.fenix.FeatureFlags
+import org.mozilla.fenix.R
+
+fun getAddonsDrawable(): Int {
+    return if (FeatureFlags.newIconSet) {
+        R.drawable.ic_proton_addons_extensions
+    } else {
+        R.drawable.ic_addons_extensions
+    }
+}
```
This is nice that we are doing it in a separate file and that we can easily clean this up when `newIconSet` is default true.

gabrielluong commented on app/src/main/res/layout/proton_component_tracking_protection_panel.xml:
```diff
@@ -0,0 +1,286 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Trying to figure out where this new layout is coming from? I don't see any details in the issue nor is it renamed from another file.

Amejia481 replied:
Thanks for catching this it looks like it's a leftover of a local iteration.

brampitoyo reviewed (changes requested):
- [x] Settings
- [x] Search
    - [x] `Find in page` looks great
    - [x] `Search engine` (above the address bar) hasn’t been replaced yet?
- [x] Tracking protection
- [x] Add-ons
    - I didn’t see the puzzle symbol updated. Maybe I didn’t checked the right place.
    - Let’s use the [`outline`](https://www.figma.com/file/UJ5BsuvpQOADF3EPyfvtyv/Icons-WIP?node-id=2244%3A1359) variant, not the `fill`.
- [x] Info
    - The `!` should be an [`i` symbol](https://www.figma.com/file/UJ5BsuvpQOADF3EPyfvtyv/Icons-WIP?node-id=2244%3A1439).
- [x] Private Browsing
    - I also wasn’t able to see the updated mask symbol in the video or screenshot
- [x] Bookmark
    - [x] Bookmarks menu fill (star with line)
    - [x] Bookmark outline (star)
    - There are two symbols here: `bookmarks menu` (star with line) and `bookmark` (star)
    - For `bookmarks menu`, let’s use the [`fill`](https://www.figma.com/file/UJ5BsuvpQOADF3EPyfvtyv/Icons-WIP?node-id=2326%3A54) variant
    - For the `Add bookmark` action, there are two separate variants:
        - [`outline`](https://www.figma.com/file/UJ5BsuvpQOADF3EPyfvtyv/Icons-WIP?node-id=2326%3A16) is already delivered
        - `fill` has not been delivered yet

Amejia481 commented:
I updated the list of icons and the video, as there are many icons changing, I preferred to send @brampitoyo a custom APK where he can see each icon in detail.   

Some of this icons are coming from this AC PR https://github.com/mozilla-mobile/android-components/pull/10132

@gabrielluong I changed my strategy about updating  icons as it was becoming a nightmare to update all possible occurrences in code and XML, I prefer to update the icons in the XML source file as it's expected that all icons are provided before the new Beta, if we don't want to ship with these new icons we just have to undo this PR, additionally changing the XML files directly safe us a lot of effort later of removing unused icons.

brampitoyo commented:
Hi @Amejia481,

For the non-AC icons:

- [x] Settings
- [x] Search
    - [x] `Find in page`
    - [x] `Search engine` (above the address bar)
- [x] Tracking protection
- [x] Add-ons
- [x] Info
- [x] Private Browsing
- [x] Bookmark
    - [x] Bookmarks menu fill (star with line)
    - [x] Bookmark outline (star)

I found just a few issues left:

- [ ] Close
    - [ ] `Tabs Tray`: still uses the old variant
    - [ ] `Find in Page`: still uses the old variant
    - [ ] `Custom Tabs`: still uses the old variant
    - [x] Sub-pages (Bookmarks, History, Downloads, etc.): correct
    - [x] New Tab Page Collection: correct
- [ ] Tabs counter
    - [ ] It looks as if the number is set too low, so the top space looks bigger than the bottom. Would it be possible to nudge the number up by around half a pixel?
-----

For the AC icons, almost everything looks great!

- [x] Suggestions
- [x] Forward
- [x] Lock
- [x] Refresh
- [x] Stop
- [x] Device mobile
- [x] Device desktop
- [x] Download
- [x] Suggestions
- [x] Extensions
- [x] What is new
- [x] Private browsing
- [ ] Warning

I found a few issues left:

- [ ] Back
    - [ ] Sub-pages (Settings, Bookmarks, History, Downloads, etc.): still uses the old variant
    - [x] Everywhere else in the app (for example: Select Tabs, Rename collection): correct
- [ ] Add-on
    - [ ] Does not appear in the panel: “Downloading and verifying add-on…”
    <img width="360" src="https://user-images.githubusercontent.com/51007/116020209-f248da80-a699-11eb-9733-ea64896e48c7.png">
    - [x] Everywhere else in the app (for example: Main menu, Add-ons Manager) is looking great
- [ ] Link
    - [ ] `Fill link from clipboard`: still uses the old variant
- [ ] Reader mode
    - [ ] `Address bar`: still uses the old variant
- [ ] Sync
    - [ ] `Tabs tray`: still uses the old variant

-----

Additionally, I’ve found these icons that you’ve included but was not listed:

- [x] Sign in
    - [x] `Tabs tray`
- [ ] Add (plus symbol)
    - [x] `Add-ons Manager`
    - [ ] Everywhere else in the app: still uses the old variant

Amejia481 commented:
Thanks for the feedback :)

> 
> Tabs counter
> 
>     It looks as if the number is set too low, so the top space looks bigger than the bottom. Would it be possible to nudge the number up by around half a pixel?

Do you mean pushing the number by around half a pixel to the top?

```
 Add (plus symbol)

    Add-ons Manager: correct
    Everywhere else in the app: still uses the old variant
```

Could you share some examples?

brampitoyo commented:
@Amejia481, thanks for your super quick response!

1. Yes. I meant pushing the number by half a pixel to the top. I’m not sure why the new icon makes the space unequal, even though it looks equal in the current square icon. Hopefully moving the tab number will help.
2. I’ve attached examples of the Add (plus symbol) in https://github.com/mozilla-mobile/fenix/pull/19044#issuecomment-826465106 above.

Amejia481 commented:
```
 Back
    Sub-pages (Settings, Bookmarks, History, Downloads, etc.): still uses the old variant
```

For the back, we always have used the Android default back navigation button, it will require a bit of effort to replace it on all sub screens for this reason a created a separated ticket for addressing it https://github.com/mozilla-mobile/fenix/issues/19258

Amejia481 commented:
I provided a updated set of APKs to @brampitoyo that match this PR update.

brampitoyo commented:
I’ve checked @Amejia481’s APK and provided this update:

- All the remaining icons, other than `Back` on sub-pages, now uses the correct variant
- `Add` and `Close` have incorrect bounding box, which was my mistake. They have now been corrected on Figma and are both ready to export. ([Artboard for `Add`](https://www.figma.com/file/UJ5BsuvpQOADF3EPyfvtyv/Icons-WIP?node-id=2326%3A215). [Artboard for `Close`](https://www.figma.com/file/UJ5BsuvpQOADF3EPyfvtyv/Icons-WIP?node-id=2326%3A215).)

brampitoyo reviewed (approved):
@Amejia481 I can now confirm that `Add` and `Close` icons appear correctly. We’re good to go – thanks heaps for working on this! r+

cadeyrn commented:
It was not mentioned in the list above so I am not sure if you're aware but the add-ons icon was only replaced in the menu of an active tab, not in the main menu of the home screen. The bookmarks icon is different as well in both menus.

gabrielluong commented on app/src/main/res/drawable/ic_onboarding_tracking_protection.xml:
```diff
@@ -2,23 +2,29 @@
-        android:height="24dp"
-        android:viewportWidth="24"
-        android:viewportHeight="24">
+<vector
```
Is this missing?

`<?xml version="1.0" encoding="utf-8"?>`

Amejia481 replied:
No,  I just updated the vector part here you can see the raw file https://github.com/mozilla-mobile/fenix/blob/f0bc8bbd87f886671ee0d5b86dd679d4bc0f2fcb/app/src/main/res/drawable/ic_onboarding_tracking_protection.xml

gabrielluong commented on app/src/main/res/drawable/ic_tracking_protection_anim_enable.xml:
```diff
@@ -1,11 +0,0 @@
-<animated-vector
-        xmlns:android="http://schemas.android.com/apk/res/android"
-        android:drawable="@drawable/ic_tracking_protection_disabled">
-    <target android:name="strike_thru_path" android:animation="@animator/strike_thru_path_enable" />
```
I think we can also remove

- strike_thru_path_enable
- strike_thru_mask_enable

since the `ic_tracking_protection_anim_X` were the only consumers fo this.

Amejia481 replied:
Would you mind pointing where you are seeing ?

* strike_thru_path_enable
* strike_thru_mask_enable

Amejia481 replied:
This ones are being used in other files

![image](https://user-images.githubusercontent.com/773158/116289043-1cfd6680-a760-11eb-87da-0710ecfbc337.png)

![image](https://user-images.githubusercontent.com/773158/116289182-43bb9d00-a760-11eb-9170-8f708d36f0e1.png)

gabrielluong commented on app/src/main/res/drawable/ic_tracking_protection_enabled.xml:
```diff
@@ -3,25 +3,11 @@
+    <path
+        android:fillColor="?primaryText"
+        android:pathData="M12.0002,22C11.5342,22 11.0662,21.885 10.6462,21.668C8.4962,20.547 6.5242,18.735 5.0942,16.566C4.3852,15.493 3.8882,14.198 3.6162,12.716L2.9092,8.872C2.6692,7.563 3.2942,6.259 4.4632,5.625L10.5652,2.363C11.4632,1.88 12.5362,1.88 13.4322,2.359L19.5342,5.682C20.7082,6.313 21.3352,7.621 21.0912,8.936L20.3862,12.728C20.1152,14.201 19.6172,15.49 18.9112,16.56C17.4822,18.729 1 ...
+</vector>
```
Add a new line EOF

Amejia481 replied:
Addressed!

gabrielluong commented on app/src/main/res/layout/quicksettings_tracking_protection.xml:
```diff
@@ -1,38 +0,0 @@
-
-    <Switch
-        android:id="@+id/trackingProtectionSwitch"
-        style="@style/QuickSettingsText.Icon"
```
Can you check if we still need these @style, @dimen, @string, and @drawable?

gabrielluong commented on app/src/main/res/drawable/ic_desktop.xml:
```diff
@@ -8,5 +8,5 @@
     <path android:fillColor="?primaryText"
-        android:pathData="M4,6a1,1 0,0 1,1 -1h14a1,1 0,0 1,1 1v11h1.5a0.5,0.5 0,0 1,0.5 0.5v1a0.5,0.5 0,0 1,-0.5 0.5h-19a0.5,0.5 0,0 1,-0.5 -0.5v-1a0.5,0.5 0,0 1,0.5 -0.5L4,17L4,6zM10,17v1h4v-1h-4zM6,7v9h12L18,7L6,7z" />
+        android:pathData="M20.5,17h-2a2.5,2.5 0,0 0,2.5 -2.5v-9A2.5,2.5 0,0 0,18.5 3h-13A2.5,2.5 0,0 0,3 5.5v9A2.5,2.5 0,0 0,5.5 17h-2a1.5,1.5 0,0 0,0 3h17a1.5,1.5 0,0 0,0 -3zM5.3,15.5l-0.8,-0.8L4.5,5.3l0.8,-0.8h13.4l0.8,0.8v9.4l-0.8,0.8L5.3,15.5zM13.25,18.5h-2.5a0.75,0.75 0,0 1,0 -1.5h2.5a0.75,0.75 0,0 1,0 1.5z" />
 </vector>
```
nit: New line at EOF

Amejia481 replied:
We could address this in another ticket, as it's not related to this PR (I didn't create the file I updated just the portion on the diff) , and there we can discuss if we want to change that globally.

Amejia481 replied:
Filed https://github.com/mozilla-mobile/fenix/issues/19273

gabrielluong commented on app/src/main/res/drawable/ic_fab_sync.xml:
```diff
@@ -1,16 +1,13 @@
+    android:viewportHeight="24">
+    <path
+        android:pathData="M12 4.675V2.501a0.5 0.5 0 0 0-0.852-0.355L8.295 4.97v0.926l2.853 2.825A0.5 0.5 0 0 0 12 8.366V6.175a5.823 5.823 0 0 1 4.434 9.602 0.75 0.75 0 0 0 1.14 0.974 7.327 7.327 0 0 0 1.75-4.75c0-4.04-3.285-7.326-7.324-7.326zm0 10.959v2.191A5.832 5.832 0 0 1 6.175 12c0-1.372 0.486-2.704 1.369-3.751a0.75 0.75 0 0 0-1.147-0.967A7.328 7.328 0 0 0 4.675 12c0 4.039 3.286 7.325 7.325 7 ...
+        android:fillColor="@color/mozac_ui_icons_fill" />
```
@color/mozac_ui_icons_fill - Nice, this is good to know

gabrielluong commented on app/src/main/res/drawable/ic_private_browsing.xml:
```diff
@@ -9,5 +9,5 @@
     android:viewportHeight="24">
     <path
         android:fillColor="@color/photonWhite"
```
This is also equivalent to mozac_ui_icons_fill. Makes you wonder if we should do something to replace all of these android:fillColor="@color/photonWhite" with mozac_ui_icons_fill. Maybe file an issue?

Amejia481 replied:
We could address this in another ticket, as it's not related to this PR, and there we can discuss if we want to change that globally.

Amejia481 replied:
I filed https://github.com/mozilla-mobile/android-components/issues/10164 for continuing the discussion

gabrielluong commented on app/src/main/res/drawable/ic_settings.xml:
```diff
@@ -8,6 +8,9 @@
-      android:pathData="M20.8 11h-2.9a6 6 0 0 0-1-2.5l2-2a1 1 0 0 0-1.4-1.4l-2 2a6 6 0 0 0-2.5-1V3.3a1 1 0 0 0-2 0V6a6 6 0 0 0-2.5 1l-2-2a1 1 0 0 0-1.4 1.4l2 2a6 6 0 0 0-1 2.5H3.3a1 1 0 0 0 0 2H6a6 6 0 0 0 1 2.5l-0.2 0.2-1.8 1.8a1 1 0 1 0 1.4 1.4L8.3 17l0.2-0.2a6 6 0 0 0 2.5 1v2.9a1 1 0 0 0 2 0v-2.9a6 6 0 0 0 2.5-1l2 2a1 1 0 0 0 1.4-1.4l-2-2a6 6 0 0 0 1-2.5h2.9a1 1 0 0 0 0-2zM8 12a4 4 0 1 1 8 0 4 ...
-      android:fillColor="?primaryText" />
+      android:pathData="M12.8,22h-1.6a2,2 0,0 1,-1.9 -1.4l-0.5,-1.8 -1,-0.6 -1.9,0.4a2,2 0,0 1,-2.2 -1L3,16.4a2,2 0,0 1,0.3 -2.4l1.3,-1.3a6.7,6.7 0,0 1,0 -1.2l-1.3,-1.3A2,2 0,0 1,3 7.7l0.8,-1.4a2,2 0,0 1,2.2 -1l1.9,0.5 1,-0.6 0.5,-1.8a2,2 0,0 1,2 -1.4h1.5a2,2 0,0 1,2 1.4l0.4,1.8 1,0.6 1.9,-0.4a2,2 0,0 1,2.2 1l0.8,1.3a2,2 0,0 1,-0.3 2.4l-1.3,1.3a6.6,6.6 0,0 1,0 1.2l1.3,1.3c0.6,0.7 0.7,1.7 0.3,2.4l ...
+      android:fillColor="?primaryText"/>
```
I also wonder if primaryText is equivalent to mozac_ui_icons_fill. Maybe note this as a potential investigation in the above issue.e

Amejia481 replied:
No it's not equivalent, [primaryText](
https://github.com/mozilla-mobile/fenix/blob/8c0f86bd96f7eae4d5cfb7af09eaa770aff06837/app/src/main/res/values/attrs.xml#L7) is an attribute (variable) that we use for changing colors for theming ([dark/light](https://github.com/mozilla-mobile/fenix/blob/8c0f86bd96f7eae4d5cfb7af09eaa770aff06837/app/src/main/res/values/styles.xml#L42)) purpose, you can find more info [here](https://developer.android.com/guide/topics/ui/look-and-feel/darktheme).

gabrielluong commented on app/src/main/res/drawable/ic_tracking_protection_disabled.xml:
```diff
@@ -7,21 +7,10 @@
-    </group>
+    <path
+        android:pathData="M3.53,3.22C3.237,2.927 2.762,2.927 2.469,3.22C2.176,3.513 2.176,3.988 2.469,4.281L4.069,5.881C3.158,6.571 2.698,7.719 2.909,8.872L3.617,12.717C3.888,14.198 4.386,15.493 5.095,16.567C6.525,18.736 8.496,20.547 10.647,21.669C11.066,21.885 11.534,22 12,22C12.468,22 12.937,21.885 13.355,21.667C14.687,20.972 15.944,20.007 17.051,18.862L19.47,21.281C19.616,21.427 19.808,21.501  ...
+        android:fillColor="@color/disabled_text"/>
```
We should perhaps create a separate variable for this? Seems weird to have the text color be unexpectedly used for icon fill colors. Can also file this as a followup

Amejia481 replied:
Sure https://github.com/mozilla-mobile/android-components/issues/10165

brampitoyo commented:
@cadeyrn I saw your feedback about missing icons in my inbox, but couldn’t see it here. I hope it was fixed, or at least we didn’t miss it!

cadeyrn commented:
@brampitoyo When I wrote my comment I thought the icons were already replaced but only the new icons from AC were already there, this PR was not yet merged. When I noticed this I removed my comment.

I checked again. The bookmarks icon is still different in both menus (home screen and active tab). Otherwise it looks good!
